startup.py

```diff
@@ -0,0 +1,238 @@
+        "houdinicore": "Houdini Core",
+        "houdinifx": "Houdini FX",
+        "hindie": "Houdini Indie",
+    }
+
+    # Glob strings to insert into the executable template paths when globbing
+    # for executables and bundles on disk. Globbing is admittedly limited in
+    # terms of specific match strings, but if we need to introduce more precise
+    # match strings later, we can do it in one place rather than each of the
+    # template paths defined below.
+    COMPONENT_GLOB_LOOKUP = {
+        "version": "*",
+        "product": "*",
+        "executable": "*",
+        "version_back": "*",
+    }
+
+    # Named regex strings to insert into the executable template paths when
+    # matching against supplied versions and products. Similar to the glob
+    # strings, these allow us to alter the regex matching for any of the
+    # variable components of the path in one place
+    COMPONENT_REGEX_LOOKUP = {
+        "version": "(?P<version>[\d.]+)",
+        "product": "(?P<product>[\w\s]+)",
+        "executable": "(?P<executable>[\w]+)",
+        "version_back": "(?P=version)",
+    }
+
+    # This dictionary defines a list of executable template strings for each
+    # of the supported operating systems. The templates are used for both
+    # globbing and regex matches by replacing the named format placeholders
+    # with an appropriate glob or regex string. As Side FX adds modifies the
+    # install path on a given OS for a new release, a new template will need
+    # to be added here.
+    # TODO: check hou 16 on windows/linux
+    EXECUTABLE_MATCH_TEMPLATES = {
+        "darwin": [
+            # /Applications/Houdini 15.5.565/Houdini.app
+            "/Applications/Houdini {version}/{product}.app",
+
+            # /Applications/Houdini/Houdini16.0.504.20/Houdini Core 16.0.504.20.app
+            "/Applications/Houdini/Houdini{version}/{product} {version_back}.app",
+        ],
+        "win32": [
+            # C:\Program Files\Side Effects Software\Houdini 15.5.565\bin\houdinifx.exe
+            "C:/Program Files/Side Effects Software/Houdini {version}/bin/{executable}.exe",
+        ],
+        "linux2": [
+            # example path: /opt/hfs14.0.444/bin/houdinifx
+            "/opt/hfs{version}/bin/{executable}",
+        ]
+    }
+
+    @property
+    def minimum_supported_version(self):
+        """The minimum supported Houdini version."""
+        return "12.0"
+
+    def scan_software(self):
+        """
+        Performs a scan for software installations.
+
+        :returns: List of :class:`SoftwareVersion` instances
+        """
+
+        software_versions = []
+
+        for sw_version in self._find_software_versions():
+            if self.is_version_supported(sw_version):
+                self.logger.debug("Accepting %s", sw_version)
+                software_versions.append(sw_version)
+            else:
+                self.logger.debug("Rejecting %s", sw_version)
```

Comment:
The `is_version_supported` gives a specific log message just before these show up in the log.